modules/add_credit/src/Plugin/Field/FieldWidget/UnitPriceRangeDefaultWidget.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * @file
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_m10n_add_credit\Plugin\Field\FieldWidget;
+
+use CommerceGuys\Intl\Formatter\CurrencyFormatterInterface;
+use Drupal\commerce_order\Plugin\Field\FieldWidget\UnitPriceWidget;
+use Drupal\commerce_price\Price;
+use Drupal\commerce_product\Entity\ProductVariationInterface;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'unit_price_range_default' widget.
+ *
+ * @FieldWidget(
+ *   id = "unit_price_range_default",
+ *   label = @Translation("Unit price (range)"),
+ *   field_types = {
+ *     "commerce_price",
+ *   }
+ * )
+ */
+class UnitPriceRangeDefaultWidget extends WidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element = [];
+
+    $element['amount'] = [
+      '#type' => 'commerce_price',
+      '#title' => $this->t('Custom amount'),
+      '#available_currencies' => array_filter($this->getFieldSetting('available_currencies')),
```

Comment:
If the currency is set on the variation, do we still need to show the currency here?

Reply:
Yes. If you have more than 1 currency, it shows as a dropdown and user can pick which one to use.